### PR TITLE
Fix (partial) #4878: option to stop autodeclaring axiom as instance.

### PR DESCRIPTION
**Pull request by SkySkimmer**
Let's see what travis does.

### CODE REVIEW

File: vernac/command.ml [hunk 1]
```diff
@@ -203,7 +214,7 @@ match local with
   let () = maybe_declare_manual_implicits false gr imps in
   let () = Universes.register_universe_binders gr pl in
   let () = assumption_message ident in
-  let () = Typeclasses.declare_instance None false gr in
+  let () = if !axiom_into_instance then Typeclasses.declare_instance None false gr in
```
Comment (SkySkimmer) [inline vernac/command.ml]: Possibly we should ignore the option and always declare instance when Discharge (Variable outside sections).

Comment (SkySkimmer) [reply vernac/command.ml]: >Possibly we should ignore the option and always declare instance when Discharge (Variable outside sections).

Being able to take Variable/Context outside sections without breaking the instances is useful so yes.

**Review by mattam82 — APPROVED** [review]
LGTM, the behavior of Variable outside of a section should change (maybe be deprecated in 8.8) but that’s orthogonal.

**Comment by maximedenes** [comment]
> the behavior of Variable outside of a section should change

What is it going to become?

**Comment by maximedenes** [comment]
@SkySkimmer I didn't know that axiom was declaring instances like that. Was it documented somewhere? I saw you added an entry for the option, but I wonder if the behavior of `Axiom` was described already.

**Comment by mattam82** [comment]
I think we discussed deprecating and removing it (maybe only Context and
not Variables), as it is redundant with Axiom, and e.g the curly braces one
can use in Context declarations make no sense at top level.
Le lun. 20 nov. 2017 à 11:18, Maxime Dénès <notifications@github.com> a
écrit :

**Comment by SkySkimmer** [comment]
>@SkySkimmer I didn't know that axiom was declaring instances like that. Was it documented somewhere? I saw you added an entry for the option, but I wonder if the behavior of Axiom was described already.

The referenced bug says it's undocumented.

**Comment by maximedenes** [comment]
HoTT fails it seems.

**Comment by SkySkimmer** [comment]
I messed up the PR number in the overlay.
Now they've merged the compatibility patch so no need for an overlay.

**Comment by maximedenes** [comment]
Rebuilding HoTT. Equations seems to have failed too, rebuilding as well.

**Comment by SkySkimmer** [comment]
For HoTT see HoTT/HoTT#909.
For equations who knows, I just rebased and can't reproduce so maybe it was just some automatic merge issue.

**Comment by SkySkimmer** [comment]
See mattam82/Coq-Equations#31.

**Comment by maximedenes** [comment]
closing and reopening to rebuild Equations

**Comment by maximedenes** [comment]
@JasonGross AppVeyor show another example of failing `coq_makefile` timing tests. Can you tell me if it is spurious? If yes, would you mind fixing it? Thanks!

**Comment by ejgallego** [comment]
I think you may want to provide a backwards compatible path for VST too.

**Comment by ejgallego** [comment]
Also note @tchajed comment's, I fully agree with him.

**Comment by maximedenes** [comment]
@ejgallego I can't stay online right now, can you open a blocker issue in the 8.8.0 milestone asking to put a CHANGES entry and provide a fix? Thanks!

**Comment by ejgallego** [comment]
#7053 is already opened , so milestoned. I am not sure what the proper fix is.

**Comment by Zimmi48** [comment]
This was already fixed! This is mentioned in CHANGES.

**Comment by Zimmi48** [comment]
And I don't see why we should fix their stuff. They pushed code that was broken with master!

**Comment by SkySkimmer** [comment]
>And I don't see why we should fix their stuff. They pushed code that was broken with master!

This is the wrong issue to talk about that.

**Pull request merged**